python_ta/reporters/html_reporter.py

```diff
@@ -91,9 +104,10 @@ def display_messages(self, layout: BaseLayout) -> None:
             self.out.flush()
         else:
             rendered_template = rendered_template.encode("utf8")
-            open_html_in_browser(
-                rendered_template, self.linter.config.watch, self.linter.config.server_port
-            )
+            if self.linter.config.watch:
+                self.persistent_server.start_server_once(rendered_template)
+            else:
+                open_html_in_browser(rendered_template, self.linter.config.watch, self.port)
```

Comment:
Given the split in code, I think you can drop the `watch` argument to `open_html_in_browser`.